Tuna corrupt configs - Type Annotation

technicalgrp89 opened the pull request:
Corrupt_configs.py , Type Annotations added. Tested the changes using mypy command
utils.groovy is updated for mypy execution path.

technicalgrp89 committed:
-- type annotations enabled
         -- pylint formatting correction
         -- mypy command added into utils.groovy

technicalgrp89 committed:
- utils.groovy mypy cmd , flags added
    - variable initialization corrected
    - ypaf formatting fix on corrupt_configs.py after develop branch merge
      local branch

    - Merge branch 'develop' of https://github.com/ROCmSoftwarePlatform/MITunaX into TypeHints_Jan25_CorrDmigrate

technicalgrp89 commented:
@erb @alexandraBara @JehandadKhan let me know

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
 
-  table_cols_conv_invmap = {v: k for k, v in TABLE_COLS_CONV_MAP.items()}
-  table_cols_fusion_invmap = {v: k for k, v in TABLE_COLS_FUSION_MAP.items()}
+  idx_1: Dict[Any, Any]
+  idx_2: Dict[Any, Any]
+  table_cols_conv_invmap: dict = {
+      idx_1: idx_2 for idx_2, idx_1 in TABLE_COLS_CONV_MAP.items()
+  }
+  table_cols_fusion_invmap: dict = {
+      idx_1: idx_2 for idx_2, idx_1 in TABLE_COLS_FUSION_MAP.items()
+  }
```
TABLE_COLS_CONV_MAP is a dict of tuples.
So calling items on this you would get (key, tuple)
here idx_2 = key
and idx_1 = tuple

technicalgrp89 replied:
idx_1  to key_idx and idx_2 to tuple_idx , more readability

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
+  cur: Any
   with open(file_name, "w") as outfile:  # pylint: disable=unspecified-encoding
-    sub_cmd_idx = None
+    sub_cmd_idx: Union[Any, slice] = None
```
sub_cmd_idx is an int

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
+  }
 
+  outfile: TextIOWrapper
+  cur: Any
```
If the type is Any, is there a benefit to giving a type?

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
       cur.execute(
-          "select config.* from job inner join config on config.id = job.config where \
-              job.arch = %s and reason = 'corrupt';", (arch,))
+          "select conv_config.* from job inner join conv_config on conv_config.id = \
```
There is no job table and conv_job doesn't have an arch column.

technicalgrp89 replied:
For this change, I am going to raise a saparate PR. In that , this querry would be re written as per the present table structures.

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
+      count: int = 0
 
+      row: str
+      sub_cmd: Any
```
sub_cmd is a string

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
-      count = 0
+      count: int = 0
 
+      row: str
```
row is a list returned by the query cursor for the current query line.

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -28,17 +28,17 @@
   """Setting up logger"""
-  logger = logging.getLogger('CorruptConfigs')
-  hdlr = logging.FileHandler('corrupt_configs.log')
+  logger: logging.Logger = logging.getLogger('CorruptConfigs')
```
It's lines like this that make me question this explicit type syntax. It's more verbose, but not any more enlightening. Does make for a good tongue twister though. @JehandadKhan

technicalgrp89 committed:
--- PR #847
         --- merge changes commit
         --- review comments addressed
         --- Dict type redefined
         --- Variable initializations
         --- Any type removed
         --- TABLE_COLS_CONV_MAP index intialized to [0]
         --- TABLE_COLS_FUSION_MAP index initalized to [0]
         --- type hints - local table variables correction
         --- variable name change - in table key:value index
         --- logger format changed to specify %d, count

technicalgrp89 commented:
@cderb , review comments addressed. Let me know.

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,51 @@ def get_logger():
 
-  table_cols_conv_invmap = {v: k for k, v in TABLE_COLS_CONV_MAP.items()}
-  table_cols_fusion_invmap = {v: k for k, v in TABLE_COLS_FUSION_MAP.items()}
+  tuple_idx: Dict[Any, Any]
+  key_idx: Dict[Any, Any]
```
Types for these are incorrect.
TABLE_COLS_CONV_MAP.items() will return an array of:
str, (str, int)
names tuple_idx and key_idx are misleading, items() returns key, value pairs

technicalgrp89 replied:
Made the changes

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,51 @@ def get_logger():
+  cur: mysql.connector.cursor.MySQLCursor
   with open(file_name, "w") as outfile:  # pylint: disable=unspecified-encoding
-    sub_cmd_idx = None
+    sub_cmd_idx: int = 0
```
Looks like it would be okay to remove this and add the type hint on line 84 where the value is initialized.

technicalgrp89 replied:
Exsiting source code : we see , line 62 -  sub_cmd_idx = None initialised here. I moved it to line where it is getting initialized.

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,51 @@ def get_logger():
-      count = 0
+      count: int = 0
 
+      row: List = []
+      sub_cmd: str
```
Possible to move these hints to the location of their previous declarations?
sub_cmd is at 90 within the loop scope.
Not sure if you can use type hints within loop definitions.
Looks like setting empty list ( = [] ) wouldn't be necessary. There are no values for type hints on lines 92 - 94.

technicalgrp89 replied:
moved declarations of count,row,sub_cmd to before using with, for loop, 92-94 idx,fds,arg_name are initalized to 0 and ""

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
 
-  table_cols_conv_invmap = {v: k for k, v in TABLE_COLS_CONV_MAP.items()}
-  table_cols_fusion_invmap = {v: k for k, v in TABLE_COLS_FUSION_MAP.items()}
+  tuple_idx: Dict[str, Tuple[str, Union[int, str]]]
```
col_tup: Tuple[str, Union[int, str]]

technicalgrp89 replied:
done

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
-  table_cols_conv_invmap = {v: k for k, v in TABLE_COLS_CONV_MAP.items()}
-  table_cols_fusion_invmap = {v: k for k, v in TABLE_COLS_FUSION_MAP.items()}
+  tuple_idx: Dict[str, Tuple[str, Union[int, str]]]
+  key_idx: Dict[str, Tuple[str, Union[int, str]]]
```
short_arg: str

technicalgrp89 replied:
done

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
-  table_cols_fusion_invmap = {v: k for k, v in TABLE_COLS_FUSION_MAP.items()}
+  tuple_idx: Dict[str, Tuple[str, Union[int, str]]]
+  key_idx: Dict[str, Tuple[str, Union[int, str]]]
+  table_cols_conv_invmap: dict = {
+      tuple_idx[0]: key_idx
+      for key_idx, tuple_idx in TABLE_COLS_CONV_MAP.items()
+  }
```
```py
table_cols_conv_invmap: Dict[str, str] = {
    col_tup[0]: short_arg
    for short_arg, col_tup in TABLE_COLS_CONV_MAP.items()
}
```

technicalgrp89 replied:
ok

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -47,31 +47,50 @@ def get_logger():
+      tuple_idx[0]: key_idx
+      for key_idx, tuple_idx in TABLE_COLS_CONV_MAP.items()
+  }
+  table_cols_fusion_invmap: dict = {
+      tuple_idx: key_idx
+      for key_idx, tuple_idx in TABLE_COLS_FUSION_MAP.items()
+  }
```
```py
table_cols_fusion_invmap: Dict[str, str] = {
    col_tup[0]: short_arg
    for short_arg, col_tup in TABLE_COLS_FUSION_MAP.items()
}
```

cderb commented on tuna/corrupt_configs.py:
```diff
@@ -85,7 +106,7 @@ def main():
         count += 1
 
-  LOGGER.warning(f"Added {count} entries")
+  LOGGER.warning("Added {count} entries")
```
The f before the string here is a formatting syntax, it would replace {count} with a value. This should be re-added.

technicalgrp89 replied:
LOGGER.warning("Entries Added = %d", count) , this address the W1203 - pylint error

technicalgrp89 committed:
develop branch merge & merge conflict on utils.groovy resolved

technicalgrp89 committed:
mypy path corrections
